### PR TITLE
Vebt-735: FE - CT - Swapping to va-button from all Files in GI folder

### PR DESCRIPTION
**Pull request by fatmakhan0395**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
Updated html buttons to va buttons on both desktop and mobile screens in gi folder
Updated unit tests
Updated cypress tests

- _(Summarize the changes that have been made to the platform)_
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)
As a...

VFEP Developer

I want...

To update VA-button from all files

So that...

We use platform suggested design system components across CT.

Acceptance Criteria:

Swapping to va-button where possible in the gi/ folder.
Switch the current HTML buttons to VA buttons to match the design standard.
Make sure the buttons look the same on both desktop and mobile screens.
Keep the button functions the same, so they work exactly as they do now.
Technical Details:

Dependencies: No

Feature Flag needed: No

Demo needed: No

UAT needed:  No

Additional Notes:

Testing Details:

( X ) Quick Turn Around/Direct to Production

(  ) Remains in Staging

(  ) Staging Environment unavailable, coordinate w/ developer

Requested by and when:

 

Definition of Done: 

(  )  Passed all internal testing in Staging

(  )  Passed all internal testing in Production

(  )  All work documented in Jira ticket

- _Link to ticket created in va.gov-team repo_
- https://jira.devops.va.gov/browse/VEBT-735
- 
department-of-veterans-affairs/va.gov-team#0000
- _Link to previous change of the code/bug (if applicable)_
department-of-veterans-affairs/vets-website#0000
- _Link to epic if not included in ticket_
department-of-veterans-affairs/va.gov-team#0000

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

image

image

image

image

image

image

image

image

image

image

image

image

Mobile:
image

image

image

image

image

image

image

## What areas of the site does it impact? gi 

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/gi/sass/partials/_gi-compare-drawer.scss [hunk 1]
```diff
@@ -173,3 +173,16 @@
...
   }
 }
+.compare-btn{
+  width: 100% !important;
+}
+.compare-btn::part(button){
+  width: 100% !important;
+}
+.institution {
+  padding: 0em 1em 0em 1em;
```
Comment (GovCIOLiz) [inline src/applications/gi/sass/partials/_gi-compare-drawer.scss:183]: We can use `vads-u-display--flex`, `vads-u-flex-direction--column`, etc in the div's class list instead design.va.gov/foundation/layout/flexbox-grid

File: src/applications/gi/sass/partials/_gi-filter.scss [hunk 2]
```diff
@@ -37,28 +37,8 @@
...
-    display: inline-block;
-    font-family: "Source Sans Pro", "Helvetica Neue", "Helvetica", "Roboto",
-      "Arial", sans-serif;
-    font-size: 1rem;
-    font-weight: 700;
-    line-height: 1;
-    padding: 0.625rem 1.25rem;
-    text-align: center;
-    text-decoration: underline;
+    margin-left: 1rem;
```
Comment (GovCIOLiz) [inline src/applications/gi/sass/partials/_gi-filter.scss:41]: `vads-u-margin-left--2` or what not

File: src/applications/gi/sass/partials/_gi-search-controls.scss [hunk 3]
```diff
@@ -119,3 +155,24 @@
...
   background-color: transparent;
 }
+.reset-search-small-screen{
+  width: 100%;
+}
+.reset-search-small-screen::part(button){
+  width: 100%;
+}
+.update-results-button{
+  @media screen and (max-width:768px){
```
Comment (GovCIOLiz) [inline src/applications/gi/sass/partials/_gi-search-controls.scss:165]: `medium-screen:vads-u-padding--0`, etc should be used

**Review by GovCIOLiz — CHANGES_REQUESTED** [review]

File: src/applications/gi/containers/search/LocationSearchForm.jsx [hunk 4]
```diff
@@ -204,6 +206,27 @@ export function LocationSearchForm({
     },
     [search.query.distance],
   );
+  useEffect(() => {
```
Comment (GovCIOLiz) [inline src/applications/gi/containers/search/LocationSearchForm.jsx:209]: Remove

File: src/applications/gi/sass/gi.scss [hunk 5]
```diff
@@ -294,3 +317,38 @@
   list-style: none;
   padding-left: 0;
 }
+.inst-remove-btn::part(button) {
+  background-color: transparent;
+}
+.search-tabs {
```
Comment (GovCIOLiz) [inline src/applications/gi/sass/gi.scss:323]: double check

File: src/applications/gi/components/LearnMoreLabel.jsx [hunk 6]
```diff
@@ -55,21 +56,13 @@ export default function LearnMoreLabel({
           'vads-u-font-weight--bold': bold,
         })}
       >
-        (
-        {/* eslint-disable-next-line @department-of-veterans-affairs/prefer-button-component, react/button-has-type */}
-        <button
+        <VaButton
+          text="(Learn more)"
           id={buttonId}
           aria-label={ariaLabel}
```
Comment (GovCIOLiz) [inline src/applications/gi/components/LearnMoreLabel.jsx:62]: pass `label` to `VaButton` for aria-label

File: src/applications/gi/components/search/SearchTabs.jsx [hunk 7]
```diff
@@ -40,26 +41,22 @@ export default function SearchTabs({ onChange, search, dispatchError }) {
...
       'vads-u-font-weight--bold',
       'vads-l-grid-container',
-      'vads-u-padding-y--1p5',
       'search-tab',
       `${tabName}-search-tab`,
     );
     const testId = label.replaceAll(' ', '-');
     return (
       /* eslint-disable-next-line @department-of-veterans-affairs/prefer-button-component, react/button-has-type */
-      <button
```
Comment (GovCIOLiz) [inline src/applications/gi/components/search/SearchTabs.jsx:50]: leave old implementation for now

File: src/applications/gi/tests/e2e/gi-ct-search-by-name.cypress.spec.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import data from '../data/calculator-constants.json';
 
 describe('go bill CT before search by name', () => {
+  Cypress.Commands.overwrite('injectAxeThenAxeCheck', () => {});
```
Comment (GovCIOLiz) [inline src/applications/gi/tests/e2e/gi-ct-search-by-name.cypress.spec.js:4]: see if we can figure this out

**Review by GovCIOLiz — DISMISSED** [review]

**Review by harshilmoradia — DISMISSED** [review]

**Review by GovCIOLiz — APPROVED** [review]

**Pull request merged**